shared/utility/skill/WalkGenerator.hpp

```diff
@@ -242,64 +252,30 @@ namespace utility::skill {
         }
 
     private:
-        // ******************************** Options ********************************
-
-        /// @brief Maximum step limits in x, y, and theta.
-        Eigen::Matrix<Scalar, 3, 1> step_limits = Eigen::Matrix<Scalar, 3, 1>::Zero();
-
-        /// @brief Step height.
-        Scalar step_height = 0.0;
-
-        /// @brief Step period (in seconds). Time for one complete step.
-        Scalar step_period = 0.0;
-
-        /// @brief Ratio of the step_period where the swing foot should be at its maximum height, between [0 1].
-        Scalar step_apex_ratio = 0.0;
-
-        /// @brief Lateral distance between feet. (how spread apart the feet should be)
-        Scalar step_width = 0.0;
-
-        /// @brief Torso height.
-        Scalar torso_height = 0.0;
-
-        /// @brief Torso pitch.
-        Scalar torso_pitch = 0.0;
-
-        /// @brief Torso constant position offset [x,y,z] (in meters)
-        Eigen::Matrix<Scalar, 3, 1> torso_position_offset = Eigen::Matrix<Scalar, 3, 1>::Zero();
-
-        /// @brief Torso offset at half step period from the planted foot, at time = step_apex_ratio * step_period
-        Eigen::Matrix<Scalar, 3, 1> torso_sway_offset = Eigen::Matrix<Scalar, 3, 1>::Zero();
-
-        /// @brief Ratio of the step_period where the torso should be at its maximum sway, between [0 1].
-        Scalar torso_sway_ratio = 0.0;
-
-        /// @brief Ratio of where to position the torso relative to the next step position [x,y] (in meters)
-        Eigen::Matrix<Scalar, 3, 1> torso_final_position_ratio = Eigen::Matrix<Scalar, 3, 1>::Zero();
+        /// @brief Walk engine parameters.
+        WalkParameters p;
```

Comment:
maybe `params` or similar? Single letter vars can get annoying when searching :laughing:

Reply:
The rule I use for single letter params is that you have to be able to see their entire usage from start to end within a single code screen

Reply:
WalkGenerator.hpp is the only file it's used in, this parameter struct is used a lot on the same lines which makes the code harder to read imo which is why I wanted to have it as a single letter